pkg/repo/badgerdb/message.go

```diff
@@ -47,34 +47,90 @@ func (r *Repo) SaveMessages(ctx context.Context, serviceID uuid.UUID, messages *
 	return nil
 }
 
-func (r *Repo) LoadMessages(ctx context.Context, serviceID uuid.UUID, language language.Tag) (*model.Messages, error) {
-	messages := model.Messages{Language: language}
-
-	_, err := r.LoadService(ctx, serviceID)
-
-	switch {
-	default:
-	// noop
-	case errors.Is(err, common.ErrNotFound):
-		return &messages, nil // Empty messages.messages for this service (Not an error)
-	case err != nil:
+// LoadMessages retrieves messages from db based on serviceID and LoadMessageOpts.
+func (r *Repo) LoadMessages(ctx context.Context, serviceID uuid.UUID, opts common.LoadMessagesOpts,
+) ([]model.Messages, error) {
+	if _, err := r.LoadService(ctx, serviceID); errors.Is(err, common.ErrNotFound) {
+		return nil, nil // Empty messages.messages for this service (Not an error)
+	} else if err != nil {
 		return nil, fmt.Errorf("repo: load service: %w", err)
 	}
 
-	err = r.db.View(func(txn *badger.Txn) error {
-		item, getErr := txn.Get(getMessagesKey(serviceID, language))
-		switch {
-		default:
-			return getValue(item, &messages)
-		case errors.Is(getErr, badger.ErrKeyNotFound):
-			return nil // Empty messages.messages for this language (Not an error)
-		case getErr != nil:
-			return fmt.Errorf("transaction: get messages: %w", getErr)
+	// load all messages if language tags are not provided.
+	if len(opts.FilterLanguages) == 0 {
+		messages, err := r.loadAllMessages(serviceID)
+		if err != nil {
+			return nil, fmt.Errorf("load all messages for service '%s': %w", serviceID, err)
 		}
-	})
+
+		return messages, nil
+	}
+
+	// load messages based on provided language tags.
+	messages, err := r.loadMessagesForLangTags(serviceID, opts.FilterLanguages)
 	if err != nil {
+		return nil, fmt.Errorf("load messages for language tags: %w", err)
+	}
+
+	return messages, nil
+}
+
+// loadMessagesForLangTags returns messages for service based on provided language tags.
+func (r *Repo) loadMessagesForLangTags(serviceID uuid.UUID, langTags []language.Tag,
+) ([]model.Messages, error) {
+	messages := make([]model.Messages, 0, len(langTags))
+
+	if err := r.db.View(func(txn *badger.Txn) error {
+		for _, langTag := range langTags {
+			var msgs model.Messages
+
+			item, er := txn.Get(getMessagesKey(serviceID, langTag))
+			switch {
+			default:
+				if valErr := getValue(item, &msgs); valErr != nil {
+					return fmt.Errorf("get messages for language tag '%s': %w", langTag, er)
+				}
+
+				messages = append(messages, msgs)
+			case errors.Is(er, badger.ErrKeyNotFound):
+				return nil // Empty messages.messages for this language (Not an error)
+			case er != nil:
+				return fmt.Errorf("transaction: get messages for language tag '%s': %w", langTag, er)
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("repo: db view: %w", err)
+	}
```

Comment:
two things that personally i would change, that in my opinion could improve readability
1. Using syntax `if err := ...; err != nil`, is only hurting readability when if body is long, for those scenarios standard
```go
err := ...
if err != nil {}
```
2. As there are 3 error varriables, i would give them appropriate namings
```go
viewErr := r.db.View...
getErr := txn.Get...
valErr := getValue...
```

Reply:
hmm, maybe could replace er with txErr, other stuff seems fine.